src/amo/components/SearchPage.js

```diff
@@ -11,15 +11,19 @@ import SearchResult from './SearchResult';
 export default class SearchPage extends React.Component {
   static propTypes = {
     count: PropTypes.number,
-    lang: PropTypes.string.isRequired,
+    lang: PropTypes.string,
     loading: PropTypes.bool.isRequired,
     page: PropTypes.number,
+    params: PropTypes.object(PropTypes.shape({
+      lang: PropTypes.string.isRequired,
+    })),
     results: PropTypes.array,
     query: PropTypes.string,
   }
 
   render() {
-    const { count, lang, loading, page, query, results } = this.props;
+    const { count, loading, page, query, results } = this.props;
+    const lang = this.props.params ? this.props.params.lang : this.props.lang;
```

Comment:
I think it might be better to pull the lang from the state in the base [mapStateToProps](https://github.com/mozilla/addons-frontend/blob/2c629fcfdf23f2dab641229dee8645806f3cdafd/src/core/containers/SearchPage.js#L7) instead. Then `lang` can continue to be required here.